frappe/automation/doctype/auto_repeat/auto_repeat.py

```diff
@@ -211,6 +218,68 @@ def set_auto_repeat_period(self, new_doc):
 			new_doc.set('from_date', from_date)
 			new_doc.set('to_date', to_date)
 
+	def get_next_schedule_date(self, schedule_date, for_full_schedule=False):
+		if month_map.get(self.frequency):
+			month_count = month_map.get(self.frequency) + month_diff(schedule_date, self.start_date) - 1
+		else:
+			month_count = 0
+
+		day_count = 0
+		if month_count and self.repeat_on_last_day:
+			day_count = 31
+			next_date = get_next_date(self.start_date, month_count, day_count)
+		elif month_count and self.repeat_on_day:
+			day_count = self.repeat_on_day
+			next_date = get_next_date(self.start_date, month_count, day_count)
+		elif month_count:
+			next_date = get_next_date(self.start_date, month_count)
+		else:
+			days = self.get_days(schedule_date)
+			next_date = add_days(schedule_date, days)
+
+		# next schedule date should be after or on current date
+		if not for_full_schedule:
+			while getdate(next_date) < getdate(today()):
+				if month_count:
+					month_count += month_map.get(self.frequency)
+					next_date = get_next_date(self.start_date, month_count, day_count)
+				elif days:
+					days = self.get_days(next_date)
+					next_date = add_days(next_date, days)
```

Comment:
On line 246, `days` will not be defined unless `month_count` is falsy. Even if it is, `days` gets re-assigned another value. Doesn't seem like line 237 is required at all. and line 246 can be `else` instead of `else if`. 

Any thoughts @ruchamahabal ?

Reply:
Line 237 is for setting the immediate `next_schedule_date`. Whereas, line 246 is only evaluated if the `for_full_schedule` flag is True. This flag is used for getting the entire auto repeat schedule and not just the immediate `next_schedule_date`. So eliminating it will not work. Changed the` else if` to `else`.